reposerver/cache/cache.go

```diff
@@ -152,8 +152,8 @@ func (c *Cache) GetManifests(revision string, appSrc *appv1.ApplicationSource, c
 		return fmt.Errorf("Unable to generate hash value: %s", err)
 	}
 
-	// If the expected hash of the cache entry does not match the actual hash value...
-	if hash != res.CacheEntryHash {
+	// If cached result does not have manifests or the expected hash of the cache entry does not match the actual hash value...
+	if hash != res.CacheEntryHash || res.ManifestResponse == nil {
 		log.Warnf("Manifest hash did not match expected value, treating as a cache miss: %s", appName)
```

Comment:
Do you want to update the log message also?

Reply:
sure, it makes sense. updated .